tests/e2e/tests/pilot/util/environment.go

```diff
@@ -164,18 +175,53 @@ func (e *Environment) ToTemplateData() TemplateData {
 		CABundle:               e.CABundle,
 		RDSv2:                  e.Config.RDSv2,
 		ImagePullPolicy:        e.Config.ImagePullPolicy,
+		MultiCluster:           e.Config.ClusterRegistriesDir,
 	}
 }
 
 // Setup creates the k8s environment and deploys the test apps
 func (e *Environment) Setup() error {
-	if e.Config.KubeConfig == "" {
+	var err error
+	if e.Config.KubeConfig != "" {
+		if _, e.KubeClient, err = kube.CreateInterface(e.Config.KubeConfig); err != nil {
+			return err
+		}
+	}
+	// ClusterRegistiresDir indicates the Kubernetes cluster config should come from files versus KUBECONFIG
+	// environmental variable.  The test config can be defined to use either a single cluster or 2 clusters
+	if e.Config.ClusterRegistriesDir != "" {
```

Comment:
Maybe refactor this as a separate method - or even remove and reuse the code used by pilot to initialize ? 
It will make changes simpler.